lib/matrix.gd

```diff
@@ -236,6 +236,31 @@ DeclareAttribute( "DefaultFieldOfMatrix", IsMatrix );
 
 #############################################################################
 ##
+#A  BaseDomain( <v> )
+#A  BaseDomain( <mat> )
+##
+##  <ManSection>
+##  <Attr Name="BaseDomain" Arg='v'/>
+##  <Attr Name="BaseDomain" Arg='mat'/>
+##
+##  <Description>
+##  For a row vector <A>v</A> that is a (necessarily nonempty) plain list,
+##  the default <Ref Attr="BaseDomain"/> method returns the default ring
+##  of <A>v</A>,
+##  see <Ref Attr="DefaultRing" Label="for ring elements"/>.
+##  <P/>
+##  For a matrix <A>mat</A> that is a (necessarily nonempty) plain list,
+##  the default <Ref Attr="BaseDomain"/> method returns the value
+##  for the first row of <A>mat</A>.
+##  </Description>
+##  </ManSection>
+##
+DeclareAttribute( "BaseDomain", IsRowVector );
+DeclareAttribute( "BaseDomain", IsMatrix );
```

Comment:
Why do we declare `BaseDomain` for "classic" vectors and matrices? Won't this become redundant once/if `IsMatrix` implies `IsMatrixObj`?

Also, why are `OneOfBaseDomain` and `ZeroOfBaseDomain` then not declared for classic objects?